ebpf/tracer.go

```diff
@@ -154,15 +159,25 @@ func (t *Tracer) Stop() {
 }
 
 func (t *Tracer) GetActiveConnections(clientID string) (*Connections, error) {
-	latestConns, latestTime, err := t.getConnections()
+	t.bufferLock.Lock()
+	defer t.bufferLock.Unlock()
+
+	latestConns, latestTime, err := t.appendActiveConnections(t.buffer[:0])
```

Comment:
Can we maintain buffer locking + growing/shrinking the buffer in `appendActiveConnections`? There's no reason it needs to be exposed

Reply:
I believe that the buffer needs to be locked for the duration of the call to `t.state.Connections` since the underlying buffer is potentially being used there.

Otherwise there could be a race condition if `GetActiveConnections` is called concurrently